openvdb_points_houdini/houdini/VRAY_OpenVDB_Points.cc

```diff
@@ -198,6 +201,98 @@ struct GenerateBBoxOp {
 
 }; // GenerateBBoxOp
 
+//////////////////////////////////////
+
+
+template <typename PointDataTreeType, typename Vec3T>
+struct CreateColorFromVelocityOp {
+
+    typedef typename PointDataTreeType::LeafNodeType                LeafNode;
+    typedef typename LeafNode::IndexOnIter                          IndexOnIter;
+    typedef typename tree::LeafManager<PointDataTreeType>           LeafManagerT;
+    typedef typename LeafManagerT::LeafRange                        LeafRangeT;
+
+    CreateColorFromVelocityOp(  const size_t colorIndex,
+                                const size_t velocityIndex,
+                                const UT_Ramp& ramp,
+                                const float maxSpeed,
+                                const std::vector<Name>& includeGroups,
+                                const std::vector<Name>& excludeGroups,
+                                const bool collapseVelocityAfter)
+        : mColorIndex(colorIndex)
+        , mVelocityIndex(velocityIndex)
+        , mRamp(ramp)
+        , mMaxSpeed(maxSpeed)
+        , mUseGroups(!includeGroups.empty() || !excludeGroups.empty())
+        , mIncludeGroups(includeGroups)
+        , mExcludeGroups(excludeGroups)
+        , mCollapseVelocityAfter(collapseVelocityAfter) { }
+
+    Vec3f getColorFromRamp(const Vec3T& velocity) const{
+
+        float proportionalSpeed = velocity.length()/(mMaxSpeed);
+        if (proportionalSpeed > 1.0f) proportionalSpeed = 1.0f;
+
+        float rampVal[4];
+        mRamp.rampLookup(proportionalSpeed, rampVal);
+        return Vec3f(rampVal[0], rampVal[1], rampVal[2]);
+    }
+
+    void operator()(LeafRangeT& range) const{
+
+        for (typename LeafRangeT::Iterator leaf=range.begin(); leaf; ++leaf) {
+
+            tools::AttributeWriteHandle<Vec3f>::Ptr colorHandle =
```

Comment:
Perhaps Vec3h is sufficient for this? I'd probably suggest adding this as an input template parameter to this class and then just instantiating based on Vec3h.